scanner/indexer/indexer.go

```diff
@@ -182,15 +226,65 @@ func NewIndexer(ctx context.Context, cfg config.IndexerConfig) (Indexer, error)
 		return nil, err
 	}
 
+	// Use indexer.Ecosystems instead of the ecosystems we pass to libindex.New
+	// in case libindex.New adds any ecosystems (which it does).
+	vscnrs, err := versionedScanners(ctx, indexer.Ecosystems)
+	if err != nil {
+		return nil, err
+	}
+
+	manifestManager := manifest.NewManager(ctx, metadataStore, locker)
+	// Set any manifests indexed prior to the existence of the manifest_metadata table
+	// to expire immediately.
+	err = manifestManager.MigrateManifests(ctx, time.Now())
+	if err != nil {
+		return nil, fmt.Errorf("migrating manifests to metadata store: %w", err)
+	}
+	// Start the manifest GC.
+	go func() {
+		if err := manifestManager.StartGC(ctx); err != nil {
+			zlog.Error(ctx).Err(err).Msg("manifest GC failed")
```

Comment:
I think we should keep that one, though, because it'll help tell us something is wrong in a way that is harder to ignore than a log. At the moment, anything not migrated to the metadata table is never GC'd, and we shouldn't miss those (there is a ticket to address that). I think I'll keep that error for now, but I can add a comment saying we can consider changing it

Reply:
Linked a Jira ticket: [032edec](https://github.com/stackrox/stackrox/pull/12648/commits/032edec4b65bcc8d1852bd3b4ba09aaa5feb940d)